Zend/zend_attributes.c

```diff
@@ -146,6 +147,43 @@ ZEND_METHOD(Override, __construct)
 	ZEND_PARSE_PARAMETERS_NONE();
 }
 
+ZEND_METHOD(Deprecated, __construct)
+{
+	zend_string *message = NULL;
+	zend_string *since = NULL;
+	zend_string *property_name;
+	zval value;
+
+	ZEND_PARSE_PARAMETERS_START(0, 2)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_NULL(message)
+		Z_PARAM_STR_OR_NULL(since)
+	ZEND_PARSE_PARAMETERS_END();
+
+	property_name = ZSTR_KNOWN(ZEND_STR_MESSAGE);
+	if (message) {
+		ZVAL_STR(&value, message);
+	} else {
+		ZVAL_NULL(&value);
+	}
+	zend_update_property_ex(zend_ce_deprecated, Z_OBJ_P(ZEND_THIS), property_name, &value);
+	zend_string_release(property_name);
+
+	/* The assignment might fail due to 'readonly'. */
+	if (EG(exception)) {
+		RETURN_THROWS();
+	}
+
+	property_name = ZSTR_INIT_LITERAL("since", 0);
+	if (since) {
+		ZVAL_STR(&value, since);
+	} else {
+		ZVAL_NULL(&value);
+	}
+	zend_update_property_ex(zend_ce_deprecated, Z_OBJ_P(ZEND_THIS), property_name, &value);
+	zend_string_release(property_name);
```

Comment:
I've added the comment + check. I expect it to be completely elided in release builds, because it's effectively just an
```
if (EG(exception)) {
  return;
}
return;
```
then.

Reply:
I personally don't like adding dead code for consistency. I think an assert would be preferable.

```c
/* Assignment cannot fail due to readonly, as $message would have already done so. */
ZEND_ASSERT(!EG(exception));
```

But this is somewhat subjective, so keep whatever you prefer.